### PR TITLE
Add GitHub workflow for `markdownlint`

### PR DESCRIPTION
**Pull request by its-miroma**

### CODE REVIEW

File: .github/workflows/markdownlint.yml
Comment (modmuss50) [inline .github/workflows/markdownlint.yml]: Specify version

File: .github/workflows/markdownlint.yml
Comment (modmuss50) [inline .github/workflows/markdownlint.yml]: Specify version

File: .github/workflows/markdownlint.yml
Comment (modmuss50) [inline .github/workflows/markdownlint.yml]: Is there a good reason to only lint changed files? It would be easier to just do it all, and would also make any changes to the lint rules also apply to existing files.

Comment (its-miroma) [reply .github/workflows/markdownlint.yml]: Performance reasons mainly, and also if all files are linted contributors may have to fix issues in unrelated files

A pull request shouldn't get blocked by this action because an unrelated file is not linted

Comment (its-miroma) [reply .github/workflows/markdownlint.yml]: I guess that wouldn't be a problem if all files were linted to begin with, though

Comment (its-miroma) [reply .github/workflows/markdownlint.yml]: > would also make any changes to the lint rules also apply to existing files

See #99, there should not be such problem after that's merged

**Review by IMB11 — APPROVED** [review]

**Comment by IMB11** [comment]
LGTM, #99 has been merged.

File: .github/workflows/markdownlint.yml
Comment (modmuss50) [inline .github/workflows/markdownlint.yml]: I would just always lint the entire thing, we arent too bothered about speed here. Keep it simple.

File: .github/workflows/markdownlint.yml
Comment (modmuss50) [inline .github/workflows/markdownlint.yml:33]: What does this do? is this magically picked up by the next step?

**Review by modmuss50 — CHANGES_REQUESTED** [review]

File: .github/workflows/markdownlint.yml
Comment (IMB11) [reply .github/workflows/markdownlint.yml]: Some translations will be unlintable based on the way the translations are patched in by Crowdin, not nessecarily about speed. I agree that we should lint versioned pages if they're edited (just not translation pages from Crowdin)

**Review by IMB11 — CHANGES_REQUESTED** [review]
See comment by modmuss on linting everything.

File: .github/workflows/markdownlint.yml
Comment (its-miroma) [reply .github/workflows/markdownlint.yml:33]: To make sure that all headings use Title Case, I used a custom `markdownlint` rule.

> is this magically picked up by the next step?

See <https://github.com/its-miroma/fabric-docs/blob/e1c496ae37812f77bfa68cba9b4419af352c3af2/.markdownlint-cli2.yaml#L50-L51>

File: .github/workflows/markdownlint.yml
Comment (its-miroma) [reply .github/workflows/markdownlint.yml]: Does linting (aka modifying) translated files cause problems in Crowdin? That's my only concern

Comment (its-miroma) [reply .github/workflows/markdownlint.yml]: > I would just always lint the entire thing

Even unchanged files? See <https://github.com/FabricMC/fabric-docs/pull/101#discussion_r1597707214>

Comment (IMB11) [reply .github/workflows/markdownlint.yml]: > Does linting (aka modifying) translated files cause problems in Crowdin? That's my only concern

No, its more the other way round, the linter would error out pretty much instantly with most of the non-latin languages (chinese, etc.), I had a try and it just doesn't support them very well.

Comment (IMB11) [reply .github/workflows/markdownlint.yml]: > Even unchanged files? See https://github.com/FabricMC/fabric-docs/pull/101#discussion_r1597707214

A good point was brought up though, if we change lint rules, it'd enforce us to change the old pages :+1:

Comment (IMB11) [reply .github/workflows/markdownlint.yml]: tldr:

- Lint all files.
- Ignore anything in versions/<version>/translated/
- Ignore anything in translated/

**Review by IMB11 — DISMISSED** [review]

**Comment by IMB11** [comment]
This seems good, LGTM.

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**
